content/commands/hexpire/index.md

```diff
@@ -0,0 +1,139 @@
+group: hash
+hidden: false
+key_specs:
+- RW: true
+  begin_search:
+    spec:
+      index: 1
+    type: index
+  find_keys:
+    spec:
+      keystep: 1
+      lastkey: 0
+      limit: 0
+    type: range
+  update: true
+linkTitle: HEXPIRE
+since: 8.0.0
+summary: Set expiration for hash fields using relative time to expire (seconds)
+syntax_fmt: HEXPIRE key seconds [NX | XX | GT | LT] numfields field [field ...]
+syntax_str: seconds [NX | XX | GT | LT] numfields field [field ...]
+title: HEXPIRE
+---
+Set an expiration (TTL or time to live) on one or more fields of a given hash key. At least one field must be specified.
+After the TTL has expired, the field(s) will automatically be deleted from the hash key.
+
+Field expirations will only be cleared by commands that delete or overwrite the
+contents of the hash key fields, including [`HDEL`]({{< relref "/commands/hdel" >}}) and [`HSET`]({{< relref "/commands/hset" >}})
+commands.
+This means that all the operations that conceptually _alter_ the value stored at a hash key's field without replacing it with a new one will leave the TTL untouched.
+
+The TTL can be cleared using the [`HPERSIST`]({{< relref "/commands/hpersist" >}}) command, turning the hash key field back into a persistent field.
+
+Note that calling `HEXPIRE`/[`HPEXPIRE`]({{< relref "/commands/hpexpire" >}}) with a non-positive TTL or
+[`HEXPIREAT`]({{< relref "/commands/hexpireat" >}})/[`HPEXPIREAT`]({{< relref "/commands/hpexpireat" >}}) with a time in the past will result in the key field being
+[deleted]({{< relref "/commands/hdel" >}}) rather than expired (accordingly, the emitted [key event]({{< relref "/develop/use/keyspace-notifications" >}})
+will be `del`, not `expired`).
+
+## Options
+
+The `HEXPIRE` command supports a set of options:
+
+* `NX` -- For each specified field, set expiration only when the field has no expiration.
+* `XX` -- For each specified field, set expiration only when the field has an existing expiration.
+* `GT` -- For each specified field, set expiration only when the new expiration is greater than current one.
+* `LT` -- For each specified field, set expiration only when the new expiration is less than current one.
+
+A non-volatile field is treated as an infinite TTL for the purpose of `GT` and `LT`.
+The `GT`, `LT` and `NX` options are mutually exclusive.
+
+## Refreshing expires
+
+It is possible to call `HEXPIRE` using as argument a key that already has an
+existing TTL set.
+In this case the time to live is _updated_ to the new value.
+
+## Example
+
+```
+redis> HEXPIRE no-key 20 NX 2 field1 field2
+(nil)
+redis> HSET mykey field1 "hello" field2 "world"
+(integer 2)
+redis> HEXPIRE mykey 10 3 field1 field2 field3
+1) (integer) 1
+2) (integer) 1
+3) (integer) -2
+redis> HGETALL mykey
+(empty array)
+```
+
+## RESP2/RESP3 replies
+
+One of the following:",
+* [Array reply](../../develop/reference/protocol-spec#arrays). For each field:
```

Comment:
"field" instead of "field(s)" as this is an array reply with a different reply per field